src/features/homePage/reducer.js

```diff
@@ -0,0 +1,153 @@
+import {
+  MOVIE_SELECTED,
+  MOVIE_UPDATED,
+  MOVIES_SEARCH,
+  MOVIES_SORT
+} from './types';
+
+const initialState = {
+  initialMovies: null,
+  sortedMovies: null,
+  currentMovieId: 0,
+  searchQuery: '',
+  sortType: 'default',
+  sortedByAscend: true
+}
+
+const updateMoviesList = (movies, movie, movieIndex) => {
+  return [
+    ...movies.slice(0, movieIndex),
+    movie,
+    ...movies.slice(movieIndex + 1)
+  ];
+};
+
+const updateMovie = (movie, value, property) => {
+  let updatedMovie = {};
+
+  switch (property) {
+    case 'likes':
+      updatedMovie = {
+        ...movie,
+        likes: movie.likes + value
+      };
+
+      break;
+
+    case 'stars':
+      updatedMovie = {
+        ...movie,
+        stars: value
+      }
+
+      break;
+
+    default:
+
+      break;
+  }
+
+  return updatedMovie;
+};
+
+const search = (movies, query) => {
+  let searchQuery;
+
+  if (query.length === 0) {
+    return movies;
+  }
+
+  return movies.filter((movie) => {
+    searchQuery = new RegExp(query, 'i');
+
+    return searchQuery.test(movie.title);
+  });
+};
+
+const resetSortProperties = (currentType, type) => {
+  if (currentType !== type) {
+    return {
+      sortType: type,
+      sortedByAscend: true
+    };
+  }
+
+  return {};
+};
+
+const compareProperties = (property, a, b, ascend = true) => {
+  const firstObject = a[property];
+  const secondObject = b[property];
+  let comparison = 0;
+
+  if (firstObject > secondObject) {
+    comparison = -1;
+  }
+
+  if (firstObject < secondObject) {
+    comparison = 1;
+  }
+
+  return ascend ? comparison : (comparison * -1);
+};
+
+const updateMoviesLists = (state, payload) => {
+  const { initialMovies, sortedMovies } = state;
+  const { id, value, property } = payload;
+
+  const sortedMovieIndex = sortedMovies.findIndex((movie) => movie.id === id);
+  const initialMovieIndex = initialMovies.findIndex((movie) => movie.id === id);
```

Comment:
All Logic with movies iteraction can be refacted. I don't like the idea of using 2 arrays for movies: sortedMovies and initialMovies. In this case you need to apply updates and changes to both of them. There should be 1 main array of your movies - 1 source of trust. You can store for ex. search result in your global state instead of store it in component state. Inside render in your components you can use Array.filter which will filtered your films by search word or show all of them. So there is a field for update